sql/2023/01/critical-css-opportunity-2.sql

```diff
@@ -0,0 +1,36 @@
+# HTTP Archive query to get % of WordPress sites not having Critical CSS implementation.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/16
+SELECT
+  client,
+  total_wp_sites,
+  sites_with_critical_css,
+  (total_wp_sites - sites_with_critical_css) AS sites_without_critical_css,
+  (total_wp_sites - sites_with_critical_css) / total_wp_sites AS opportunity
+FROM (
+  SELECT
+    pages._TABLE_SUFFIX AS client,
+    COUNT(pages.url) AS total_wp_sites,
+    COUNTIF( CAST(JSON_EXTRACT_SCALAR(payload, '$._externalCssInHead') AS INT64) = 0
+      AND CAST(JSON_EXTRACT_SCALAR(payload, '$._inlineCssInHead') AS INT64) > 0
+      AND ( CAST(JSON_EXTRACT_SCALAR(payload, '$._externalCssInHead') AS INT64) + CAST(JSON_EXTRACT_SCALAR(payload, '$._externalCssinBody') AS INT64) ) > 0 ) AS sites_with_critical_css
```

Comment:
A few things need to be improved here, some more critical than others:
1. The data is available as JSON in a `_css` object within the `payload` JSON, so we need to `JSON_EXTRACT_SCALAR` calls. It's a bit similar to how it's done in https://github.com/GoogleChromeLabs/wpp-research/pull/29/files#diff-d647443f186ed55d494b9389bc8322970137af7a407a69bcc8109c370cff0001R20.
2. You have a typo in the `externalCssinBody` name, it needs to be a capital i like in the other names.
3. In line 30, I think we can get rid of the first part of the sum: We already ensure that `externalCssInHead` needs to be "0" above, so this value will always be "0", no need to include it in the calculation.
4. Maybe reformat this for better readability so that every condition is in its own line.
```suggestion
    COUNTIF(
      CAST(JSON_EXTRACT_SCALAR(JSON_EXTRACT_SCALAR(payload, '$._css'), '$.externalCssInHead') AS INT64) = 0
      AND CAST(JSON_EXTRACT_SCALAR(JSON_EXTRACT_SCALAR(payload, '$._css'), '$.inlineCssInHead') AS INT64) > 0
      AND CAST(JSON_EXTRACT_SCALAR(JSON_EXTRACT_SCALAR(payload, '$._css'), '$.externalCssInBody') AS INT64) > 0
    ) AS sites_with_critical_css
```

Reply:
Updated.